### PR TITLE
slurm: correctly configure StateSaveLocation and SlurmdSpoolDir

### PR DESCRIPTION
**Pull request by demartinofra**
`StateSaveLocation` and `SlurmdSpoolDir` were being written to /tmp dir and this was causing the following error to appear in the slurm logs other than the fact that the files might be deleted on node reboots.

```
[2020-02-24T09:31:31.586] error: ################################################
[2020-02-24T09:31:31.586] error: ###       SEVERE SECURITY VULERABILTY        ###
[2020-02-24T09:31:31.586] error: ### StateSaveLocation DIRECTORY IS WORLD WRITABLE ###
[2020-02-24T09:31:31.586] error: ###         CORRECT FILE PERMISSIONS         ###
[2020-02-24T09:31:31.586] error: ################################################
```

slurmctld logs after the fix:
```
[2020-02-24T16:37:46.536] error: chdir(/var/log): Permission denied
[2020-02-24T16:37:46.536] slurmctld version 19.05.5 started on cluster parallelcluster
[2020-02-24T16:37:48.315] No memory enforcing mechanism configured.
[2020-02-24T16:37:48.315] layouts: no layout to initialize
[2020-02-24T16:37:48.655] layouts: loading entities/relations information
[2020-02-24T16:37:48.655] error: Could not open node state file /var/spool/slurm.state/node_state: No such file or directory
[2020-02-24T16:37:48.655] error: NOTE: Trying backup state save file. Information may be lost!
[2020-02-24T16:37:48.655] No node state file (/var/spool/slurm.state/node_state.old) to recover
[2020-02-24T16:37:48.655] error: Could not open job state file /var/spool/slurm.state/job_state: No such file or directory
[2020-02-24T16:37:48.655] error: NOTE: Trying backup state save file. Jobs may be lost!
[2020-02-24T16:37:48.655] No job state file (/var/spool/slurm.state/job_state.old) to recover
[2020-02-24T16:37:48.655] select/cons_tres: select_p_node_init
[2020-02-24T16:37:48.655] select/cons_tres: preparing for 1 partitions
[2020-02-24T16:37:48.655] error: Could not open reservation state file /var/spool/slurm.state/resv_state: No such file or directory
[2020-02-24T16:37:48.655] error: NOTE: Trying backup state save file. Reservations may be lost
[2020-02-24T16:37:48.655] No reservation state file (/var/spool/slurm.state/resv_state.old) to recover
[2020-02-24T16:37:48.656] error: Could not open trigger state file /var/spool/slurm.state/trigger_state: No such file or directory
[2020-02-24T16:37:48.656] error: NOTE: Trying backup state save file. Triggers may be lost!
[2020-02-24T16:37:48.656] No trigger state file (/var/spool/slurm.state/trigger_state.old) to recover
[2020-02-24T16:37:48.656] _preserve_plugins: backup_controller not specified
[2020-02-24T16:37:48.656] Reinitializing job accounting state
[2020-02-24T16:37:48.656] select/cons_tres: select_p_reconfigure
[2020-02-24T16:37:48.656] select/cons_tres: select_p_node_init
[2020-02-24T16:37:48.656] select/cons_tres: preparing for 1 partitions
[2020-02-24T16:37:48.656] Running as primary controller
[2020-02-24T16:37:48.741] No parameter for mcs plugin, default values set
[2020-02-24T16:37:48.741] mcs: MCSParameters = (null). ondemand set.
[2020-02-24T16:37:59.661] Terminate signal (SIGINT or SIGTERM) received
[2020-02-24T16:37:59.754] Saving all slurm state
[2020-02-24T16:37:59.754] error: Could not open job state file /var/spool/slurm.state/job_state: No such file or directory
[2020-02-24T16:37:59.754] error: NOTE: Trying backup state save file. Jobs may be lost!
[2020-02-24T16:37:59.754] No job state file (/var/spool/slurm.state/job_state.old) found
[2020-02-24T16:37:59.765] layouts: all layouts are now unloaded.
[2020-02-24T16:37:59.795] error: chdir(/var/log): Permission denied
[2020-02-24T16:37:59.795] slurmctld version 19.05.5 started on cluster parallelcluster
[2020-02-24T16:37:59.797] No memory enforcing mechanism configured.
[2020-02-24T16:37:59.797] layouts: no layout to initialize
[2020-02-24T16:37:59.799] layouts: loading entities/relations information
[2020-02-24T16:37:59.799] Recovered state of 10 nodes
[2020-02-24T16:37:59.799] Recovered information about 0 jobs
[2020-02-24T16:37:59.799] select/cons_tres: select_p_node_init
[2020-02-24T16:37:59.799] select/cons_tres: preparing for 1 partitions
[2020-02-24T16:37:59.800] Recovered state of 0 reservations
[2020-02-24T16:37:59.800] _preserve_plugins: backup_controller not specified
[2020-02-24T16:37:59.800] select/cons_tres: select_p_reconfigure
[2020-02-24T16:37:59.800] select/cons_tres: select_p_node_init
[2020-02-24T16:37:59.800] select/cons_tres: preparing for 1 partitions
[2020-02-24T16:37:59.800] Running as primary controller
[2020-02-24T16:37:59.800] No parameter for mcs plugin, default values set
[2020-02-24T16:37:59.800] mcs: MCSParameters = (null). ondemand set.
[2020-02-24T16:38:00.033] Processing RPC: REQUEST_RECONFIGURE from uid=0
[2020-02-24T16:38:00.033] No memory enforcing mechanism configured.
[2020-02-24T16:38:00.033] layouts: no layout to initialize
[2020-02-24T16:38:00.034] restoring original state of nodes
[2020-02-24T16:38:00.034] select/cons_tres: select_p_node_init
[2020-02-24T16:38:00.034] select/cons_tres: preparing for 1 partitions
[2020-02-24T16:38:00.034] _preserve_plugins: backup_controller not specified
[2020-02-24T16:38:00.034] select/cons_tres: select_p_reconfigure
[2020-02-24T16:38:00.034] select/cons_tres: select_p_node_init
[2020-02-24T16:38:00.034] select/cons_tres: preparing for 1 partitions
[2020-02-24T16:38:00.034] No parameter for mcs plugin, default values set
[2020-02-24T16:38:00.034] mcs: MCSParameters = (null). ondemand set.
[2020-02-24T16:38:00.034] _slurm_rpc_reconfigure_controller: completed usec=1251
[2020-02-24T16:38:02.802] SchedulerParameters=default_queue_depth=100,max_rpc_cnt=0,max_sched_time=2,partition_job_depth=0,sched_max_job_start=0,sched_min_interval=2
[2020-02-24T16:43:17.818] _slurm_rpc_submit_batch_job: JobId=2 InitPrio=4294901759 usec=317
```

Saving files to a persistent location and setting permissions to the slurm user to access such directory.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: recipes/_compute_slurm_config.rb [hunk 1]
```diff
@@ -15,6 +15,13 @@
 # OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Create directory configured as SlurmdSpoolDir
+directory '/var/spool/slurmd.spool' do
```
Comment (enrico-usai) [inline recipes/_compute_slurm_config.rb]: Why don't use the default folder `/var/spool/slurmd` as mentioned in the [documentation](https://slurm.schedmd.com/slurm.conf.html)?

File: recipes/_master_slurm_config.rb [hunk 2]
```diff
@@ -29,6 +29,13 @@
   mode '0755'
 end
 
+# Create directory configured as StateSaveLocation
+directory '/var/spool/slurm.state' do
```
Comment (enrico-usai) [inline recipes/_master_slurm_config.rb]: Why don't use the default folder `/var/spool` as mentioned in the [documentation](https://slurm.schedmd.com/slurm.conf.html)?

Comment (enrico-usai) [reply recipes/_master_slurm_config.rb]: Or another idea could be to use: `/var/spool/slurmctld` since it's specific for the `slurmctld` daemon.

Comment (demartinofra) [reply recipes/_master_slurm_config.rb]: For this one I'd rather have a subdirectory reserved to the slurm user

File: recipes/_compute_slurm_config.rb (hunk 1 above)
Comment (demartinofra) [reply recipes/_compute_slurm_config.rb]: this one I changed it to `/var/spool/slurmd`

**Pull request merged**